plugins/_collections.js

```diff
@@ -2289,3 +2289,66 @@ exports.pseudoClasses = {
   userAction: ['hover', 'active', 'focus', 'focus-visible', 'focus-within'],
   functional: ['is', 'not', 'where', 'has'],
 };
+
+/**
+ * @type {string[]}
+ * @see https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute
+ */
+exports.deprecatedAttrs = [
```

Comment:
I'd prefer if we had these in the respective nodes, and if any of them to apply to all nodes, we can have a `deprecatedCoreAttrs` outside which is destructed into the relevant nodes.

It's because `removeUnknownsAndDefaults` already handles this for all other nodes, so I don't think there's any need to iterate through every item on _every_ node. In my opinion, it also makes more sense since it's technically only deprecated in the places it was valid before, it can't be deprecated in a node it was never supported in.

Reply:
> For example, the clip attribute should probably be migrated to the clip-path CSS property rather than removed.

This kind of setup would be unexpected for cases where the previous attribute wasn't recognized by most SVG renderers.
+ clip doesn't do the same thing as clip-path in my experience with CSS, you would need to do some complicated conversions.

Reply:
> This kind of setup would be unexpected for cases where the previous attribute wasn't recognized by most SVG renderers, but the new one is.

While that's technically valid, I think it's safe to assume that if an SVG uses `clip`, they were intending to use the feature. Plus, that argument can be made for any modification we do, though I'll concede in cases where it really does apply to the majority of clients. We'll see after looking into each migratable attribute individually.

The rest of this will cover `clip` exclusively, since that's the only one I've looked at so far.

I've only tested `clip` and `clip-path` in two clients so far, Firefox and Eye of GNOME. Both work in Firefox, and neither work in Eye of GNOME, so in both cases rendering isn't affected.

I'll need to investigate this further, but the scenario you've presented is more like a rendering bug or unimplemented feature in certain clients, in which case this would actually work around the bug and increase compatibility.

In the worst case, if it may result in changes to rendering, it wouldn't be a default plugin anyway. It's up to users to decide whether that's worthwhile for them or not based on the warning in the documentation. (Non-default plugins ideally won't, but have more leeway to make changes that impact rendering or compatibility, as long as they're logical and adhere to whatever spec they're following, usually SVG or HTML.)

> Looks like SVG's clip only accepts "rect", which could be migrated to "inset"

I just took the alternative from MDN, but didn't check further:

> Warning: This property is deprecated. Use [clip-path](https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path) instead.
> 
> — https://developer.mozilla.org/docs/Web/SVG/Attribute/clip

Here is what I tested with:

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 10">
  <svg x="0" width="10" height="10" clip="rect(1, 9, 8, 2)">
    <circle cx="5" cy="5" r="4" stroke="green" />
  </svg>
</svg>
```

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 10">
  <svg x="0" width="10" height="10" style="clip-path:inset(0px 1px 2px 1px)">
    <circle cx="5" cy="5" r="4" stroke="green" />
  </svg>
</svg>
```

---

This weekend I'll look into each attribute and their compatibility table individually, then I'll drop a line with what I think.

Meanwhile, other opinions welcome, especially if anyone else wants to do the same.